spec/requests/partners_spec.rb

```diff
@@ -0,0 +1,135 @@
+require "rails_helper"
+
+RSpec.describe "Partners", type: :request, include_shared: true do
+  let(:partner) { create(:partner) }
+  let(:user) { create(:user, partner: partner) }
+
+  context "when authenticated" do
+    before do
```

Comment:
I like to avoid `before` blocks for actions such as this for reasons similar to my preference against `let` blocks.